plugins/auth-backend-module-oidc-provider/src/authenticator.ts

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Issuer,
+  ClientAuthMethod,
+  TokenSet,
+  UserinfoResponse,
+  Strategy as OidcStrategy,
+} from 'openid-client';
+import {
+  createOAuthAuthenticator,
+  PassportOAuthAuthenticatorHelper,
+  PassportOAuthDoneCallback,
+} from '@backstage/plugin-auth-node';
+
+/** @public */
+export const oidcAuthenticator = createOAuthAuthenticator({
+  defaultProfileTransform:
+    PassportOAuthAuthenticatorHelper.defaultProfileTransform,
```

Comment:
Assuming `TContext` is that `{ helper, client, initializedScope, initializedPrompt, strategy }` type we're using elsewhere, @Rugvip made a wise suggestion: we may want `oidcAuthenticator` to be of type `OAuthAuthenticator<TContext,UserinfoResponse>`, instead of `OAuthAuthenticator<TContext,PassportProfile>` which it is now.

I think the choice of `PassportOAuthAuthenticatorHelper.defaultProfileTransform` as the default profileTransform caused the typechecker to infer this `PassportProfile` type, which forced all the munging going on down where `strategy` gets initialized.